sources/pmwebapi_test.go

```diff
@@ -0,0 +1,98 @@
+package sources
+
+import (
+	"testing"
+	"encoding/json"
+)
+
+type AutoGenerated struct {
+	Args struct {
+	} `json:"args"`
+	Headers struct {
+		Accept string `json:"Accept"`
+		AcceptEncoding string `json:"Accept-Encoding"`
+		AcceptLanguage string `json:"Accept-Language"`
+		CacheControl string `json:"Cache-Control"`
+		Connection string `json:"Connection"`
+		Host string `json:"Host"`
+		UpgradeInsecureRequests string `json:"Upgrade-Insecure-Requests"`
+		UserAgent string `json:"User-Agent"`
+		XBluecoatVia string `json:"X-Bluecoat-Via"`
+	} `json:"headers"`
+	Origin string `json:"origin"`
+	URL string `json:"url"`
+}
+
+func TestGetRequest(t *testing.T) {
+	testString := "http://httpbin.org/get"
+	response := getRequest(testString)
+	if response == nil {
+		t.Fatal("retrieved an unexpected result")
+	}
+	A := &AutoGenerated{}
+	err := json.Unmarshal(response, A)
+	if err != nil {
+		t.Fatal("fail")
+	}
+	if A.Headers.Host != "httpbin.org" {
+		t.Fatal("fail")
+	}
+}
+
+func TestUnmarshal(t *testing.T) {
+	testUrl := "http://httpbin.org/get"
+	testBody := getRequest(testUrl)
+	testMap := &AutoGenerated{}
+	unmarshal([]byte(testBody),testMap)	
+	if testMap.Headers.Host != "httpbin.org" {
+		t.Fatal("faIL")
+	}
+	if testMap.Origin == "12kk" {
+		t.Fatal("fail")
+	}
+	if testMap.URL != "http://httpbin.org/get" {
+		t.Fatal("fail")
+	}			
+}
+  
+func TestLabelTypes(t *testing.T) {
+	testStringOne := ""
```

Comment:
I'm generally not much of a stickler for variable names, but it would be helpful to have more specific names here. I had to look at the `typeLabel` function side-to-side in order to figure out what these three variables should be doing. You can even just copy the variable names used in the `typeLabel` function (`units`, `pcpType`, and `name`).